From 479a96cde7a51a6e7aead7ca4c2c0828253aa45a Mon Sep 17 00:00:00 2001
From: Yuqing Zhu <carol.zhu@nxp.com>
Date: Wed, 17 Apr 2019 18:48:21 +0800
Subject: [PATCH] linuxptp: Fix unknown type name 'clockid_t'

Error:
linuxptp/1.8-r0/recipe-sysroot/usr/include/linux/net_tstamp.h:158:2: error: unknown type name 'clockid_t'
|   clockid_t clockid; /* reference clockid */
|   ^~~~~~~~~
| In file included from clock.c:20:

net_tstamp.h in recent kernel versions requires time.h for clockid_t.

Signed-off-by: Yuqing Zhu <carol.zhu@nxp.com>
---
 clock.c       | 2 +-
 hwstamp_ctl.c | 1 +
 raw.c         | 1 +
 sk.c          | 1 +
 timemaster.c  | 1 +
 5 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/clock.c b/clock.c
index c716f01..6a509cb 100644
--- a/clock.c
+++ b/clock.c
@@ -17,11 +17,11 @@
  * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 #include <errno.h>
+#include <time.h>
 #include <linux/net_tstamp.h>
 #include <poll.h>
 #include <stdlib.h>
 #include <string.h>
-#include <time.h>
 #include <sys/queue.h>
 
 #include "address.h"
diff --git a/hwstamp_ctl.c b/hwstamp_ctl.c
index 0d21843..4df9c04 100644
--- a/hwstamp_ctl.c
+++ b/hwstamp_ctl.c
@@ -25,6 +25,7 @@
 #include <unistd.h>
 
 #include <arpa/inet.h>
+#include <time.h>
 #include <linux/net_tstamp.h>
 #include <linux/sockios.h>
 #include <net/if.h>
diff --git a/raw.c b/raw.c
index 494ea7f..ffc8053 100644
--- a/raw.c
+++ b/raw.c
@@ -33,6 +33,7 @@
 #include <linux/filter.h>
 #include <linux/if_ether.h>
 #include <linux/errqueue.h>
+#include <time.h>
 #include <linux/net_tstamp.h>
 #include <linux/sockios.h>
 
diff --git a/sk.c b/sk.c
index 63ec206..61f5538 100644
--- a/sk.c
+++ b/sk.c
@@ -18,6 +18,7 @@
  * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 #include <errno.h>
+#include <time.h>
 #include <linux/net_tstamp.h>
 #include <linux/sockios.h>
 #include <linux/ethtool.h>
diff --git a/timemaster.c b/timemaster.c
index 66ac521..09fd6fb 100644
--- a/timemaster.c
+++ b/timemaster.c
@@ -22,6 +22,7 @@
 #include <errno.h>
 #include <libgen.h>
 #include <limits.h>
+#include <time.h>
 #include <linux/net_tstamp.h>
 #include <signal.h>
 #include <spawn.h>
-- 
1.9.1

